Automattic/wp-calypso
# Lasagna: sync channel joins to new backend topics

#### Changes proposed in this Pull Request

* Swap in [lasagna.js](https://github.com/Automattic/lasagna.js)
* Adjust the frontend JS to align with backend server expectations.

Props Andrei for nailing reliable full view component load/unload and for `viewing`.

#### Testing instructions

1. Boot this branch in dev (not calypso.live).
2. Enable `lasagna:*` debug.
3. Go to Reader. Move in and out of _public post_ full post views. You should see socket connect on Reader enter, channel joins on full post view load, channel leaves on full post view exit, socket disconnect when leaving the reader, and everything still as expected on hard refreshes anywhere in that flow.
